haystack/components/agents/agent.py

```diff
@@ -296,15 +301,51 @@ def _initialize_fresh_execution(
             tool_invoker_inputs=tool_invoker_inputs,
         )
 
+    def _select_tools(
+        self, tools: Optional[Union[list[Tool], Toolset, list[str]]] = None
+    ) -> Union[list[Tool], Toolset]:
+        """
+        Select tools for the current run based on the provided tools parameter.
+
+        :param tools: Optional list of Tool objects, a Toolset, or list of tool names to use for this run.
+            When passing tool names, tools are selected from the Agent's originally configured tools.
+        :return: Selected tools for the current run.
```

Comment:
We use `:returns:` in our docstrings
```suggestion
        :returns: Selected tools for the current run.
```